core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1206,6 +1205,54 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     }
   }
 
+  private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlWriter.Frame ifFrame = writer.startFunCall("IF");
+    unParseRegexpContains(writer, call, leftPrec, rightPrec);
+    writer.sep(",");
+    writer.literal("1");
+    writer.sep(",");
+    writer.literal("0");
+    writer.endFunCall(ifFrame);
+  }
+
+  private void unParseRegexpContains(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    SqlWriter.Frame regexContainsFrame = writer.startFunCall("REGEXP_CONTAINS");
+    call.operand(0).unparse(writer, leftPrec, rightPrec);
+    writer.print(", r");
+    unParseRegexString(writer, call, leftPrec, rightPrec);
+    writer.endFunCall(regexContainsFrame);
+  }
+
+  private void unParseRegexString(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
```

Comment:
Move common code to org.apache.calcite.util.Util.java

Reply:
Inside the unParseRegexString() method, we call another method that has diff implementations for each dialect.
So It is not possible to make it common.